iati/validator.py

```diff
@@ -445,6 +446,8 @@ def _check_is_iati_xml(dataset, schema):
         for log_entry in doc_invalid.error_log:  # pylint: disable=no-member
             error = _create_error_for_lxml_log_entry(log_entry)
             error_log.add(error)
+    except AttributeError:
+        raise TypeError('The provided argument that should have been a Dataset was actually a {0}'.format(type(dataset)))
```

Comment:
Maybe better to word:
`('Unexpected argument: {0} is not an iati.data.Dataset'.format(type(dataset)))`

Reply:
https://github.com/IATI/pyIATI/pull/239/commits/7fa0dbc738e05be4c8f2b02307674d2900410536